src/components/SelectField/SelectField.tsx

```diff
@@ -0,0 +1,27 @@
+import { FieldProps } from 'formik';
+import Select from 'components/Select';
+import { FormValuesType } from 'components/HistoricalForecast/validation';
+import OptionBase from 'types/optionBase';
+
+interface Props<Option extends OptionBase>
+  extends FieldProps<Option['value'], FormValuesType> {
+  labelId: string;
+  label: string;
+  options: Option[];
+}
+
+export default function SelectField<Option extends OptionBase>({
+  field: { name, value, onChange },
+  ...props
+}: Props<Option>) {
+  return (
+    <Select
+      value={value}
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      setValue={onChange(name)}
```

Comment:
You shouldn't call `onChange`` immediately. Try to pass `name` attribute with to `Select`. `onChange` may not work because `formik` takes not only value but also `name` from `e.target`